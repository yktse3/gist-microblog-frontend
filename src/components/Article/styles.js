import styled from 'styled-components';

export const Card = styled.div`
  background-color: #f1edf1;
  padding: 20px;
  font-size: 18px;
  display: flex;
  flex-direction: column;
  height: 100%;
  justify-content: left;
`;

export const ContentBlock = styled.span`
  margin-top: 10px;
`;

export const CommentText = styled.span`
  font-size: 15px;
  color: #888481;
  margin: 10px 10px 0 auto;

  &:hover {
    cursor: pointer;
  }
`;
