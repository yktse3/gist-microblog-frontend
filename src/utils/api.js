const CLIENT_ID = 'b77a50cf0d2a3029972c';
const CLIENT_SECRET = 'f29284e62d8c8bdd0fba0b304b7c5c243eec0d88';
const HEADER_ACCEPT_JSON = 'application/json';
const HEADER_ACCEPT_GITHUB_V3 = 'application/vnd.github.v3+json';

const defaultConfig = {
  accessToken: null,
};

class API {
  constructor(config = defaultConfig) {
    this._config = config;
  }

  changeAccessToken = (accessToken) => {
    this._config.accessToken = accessToken;
  }

  getAccessToken = async ({ code }) => {
    const url = 'http://localhost:3100/login';
    const body = {
      client_id: CLIENT_ID,
      client_secret: CLIENT_SECRET,
      code,
    };

    const config = {
      method: 'POST',
      headers: {
        Accept: HEADER_ACCEPT_JSON,
        'Content-Type': HEADER_ACCEPT_JSON,
        'Access-Control-Allow-Origin': '*',
      },
      body: JSON.stringify(body),
    };

    try {
      const result = await fetch(url, config);
      const response = await result.json();
      return response;
    } catch (e) {
      return Promise.reject(e);
    }
  };
}

export default new API();
