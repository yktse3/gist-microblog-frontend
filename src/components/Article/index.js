import React, { Component } from 'react';
import {
  string,
  number,
  func,
  arrayOf,
  shape,
  int,
} from 'prop-types';
import ArticleTitle from 'components/ArticleTitle';
import CommentList from 'components/CommentList';
import NewComment from 'components/NewComment';
import { noop } from 'utils/helpers';
import {
  Card,
  ContentBlock,
  CommentText,
} from './styles';

class Article extends Component {
  static propTypes = {
    title: string.isRequired,
    dateTime: string.isRequired,
    content: string.isRequired,
    numOfComment: number.isRequired,
    commentURL: string.isRequired,
    onCommentClicked: func,
    articleID: string.isRequired,
    comments: arrayOf(
      shape({
        id: int,
        userID: string,
        content: string,
      }),
    ),
    onCommentSubmit: func.isRequired,
  };

  static defaultProps = {
    onCommentClicked: noop,
    comments: [],
  };

  state = {};

  showComments = () => {
    if (this.props.numOfComment > 0) {
      this.props.onCommentClicked(this.props.articleID, this.props.commentURL);
    }
  }

  render() {
    const {
      title,
      dateTime,
      content,
      numOfComment,
    } = this.props;

    return (
      <div>
        <Card>
          <ArticleTitle
            title={title}
            dateTime={dateTime}
          />
          <ContentBlock>{content}</ContentBlock>
          <CommentText onClick={this.showComments}>{`${numOfComment} comments`}</CommentText>
        </Card>
        {
          this.props.comments && (
            <CommentList
              comments={this.props.comments}
            />
          )
        }
        <NewComment
          onSubmit={this.props.onCommentSubmit}
          articleID={this.props.articleID}
        />
      </div>
    );
  }
}

export default Article;
