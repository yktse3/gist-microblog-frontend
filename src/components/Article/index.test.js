import React from 'react';
import renderer from 'react-test-renderer';
import { noop } from 'utils/helpers';
import Article from './index';

describe('Article', () => {
  it('render successfully', () => {
    const component = renderer.create(
      <Article
        title="A title"
        dateTime="2018/10/15 17:13:00"
        content="Content"
        numOfComment={1}
        key="1"
        articleID="1"
        commentURL="http://whatever.com/comment/1"
        onCommentClicked={noop}
        comments={[]}
        onCommentSubmit={noop}
      />,
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
