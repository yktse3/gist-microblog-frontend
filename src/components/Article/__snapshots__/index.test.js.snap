// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article render successfully 1`] = `
<div>
  <div
    className="sc-EHOje jiRcjx"
  >
    <div
      className="sc-bdVaJa bPGLpq"
    >
      <span
        className="sc-bwzfXH dNCrpK"
      >
        A title
      </span>
      <span
        className="sc-htpNat cUROdq"
      >
        2018/10/15 17:13:00
      </span>
    </div>
    <span
      className="sc-bZQynM dLZvNn"
    >
      Content
    </span>
    <span
      className="sc-gzVnrw kDPufM"
      onClick={[Function]}
    >
      1 comments
    </span>
  </div>
  <div
    className="sc-htoDjs eolgBN"
  />
  <div
    className="sc-EHOje jiRcjx"
  >
    <textarea
      autoComplete="off"
      className="sc-jTzLTM gKGFVP sc-iwsKbI iCkmLb"
      name="content"
      onChange={[Function]}
      placeholder="Comment"
      value=""
    />
    <button
      className="sc-gZMcBi bMlNzm"
      onClick={[Function]}
    >
      Submit
    </button>
  </div>
</div>
`;
